lib/ClangImporter/ClangImporter.cpp

```diff
@@ -4672,8 +4672,7 @@ clang::FunctionDecl *ClangImporter::instantiateCXXFunctionTemplate(
     ctx.Diags.diagnose(SourceLoc(),
                        diag::unable_to_convert_generic_swift_types.ID,
                        {func->getName(), StringRef(failedTypesStr)});
-    // Return a valid FunctionDecl but, we'll never use it.
-    return func->getAsFunction();
+    return nullptr;
```

Comment:
Is there a test case that tests this change? I couldn't spot one in your test cases.

Reply:
This is just moved up. There's already a test that covers this (where we verify clang's substitution failure errors).